cadquery/hull.py

```diff
@@ -292,24 +302,36 @@ def arc_arc(a1: Arc, a2: Arc) -> Tuple[float, Segment]:
         Segment(Point(x12, y12), Point(x22, y22)),
     )
 
-    return angles[ix], segments[ix]
+    segment_angle = atan2p(segments[ix].b.x - xc2, segments[ix].b.y - yc2)
+    if (segment_angle < a2.a1 or segment_angle > a2.a2):
+        return inf, Segment(Point(inf, inf), Point(inf, inf))
```

Comment:
I don't understand this function that well, so this might have been from something else wrong and is now redundant, but somehow it was selecting a point not actually on the arc, so this if statement throws that out if that happens.

I'd bet this isn't the best way to handle it and there's more edge cases that would fail here.